install_files/ansible-base/roles/restrict-direct-access/tasks/dh_moduli.yml

```diff
@@ -0,0 +1,21 @@
+---                                                                             
+- name: Check whether Diffie-Hellman groups have been updated                   
+  stat:                                                                         
+    path: /etc/ssh/moduli                                                       
+  register: dh_moduli                                                           
+
+# SHA1 checksum represents the latest default primes/moduli shipped with OpenSSH
+                                                                                
+- name: Remove weak DH moduli                                                   
+  command: awk '$5 > 2000' /etc/ssh/moduli                                      
+  register: new_dh_moduli                                                       
+  when: dh_moduli.stat.checksum == '2dc7b3b18a2d3884ca4604466b75cf7903e96a97'   
+                                                                                
+- name: Install updated DH moduli                                               
+  copy:                                                                         
+    dest: /etc/ssh/moduli                                                       
+    content: "{{ new_dh_moduli.stdout }}"                                       
+    owner: root                                                                 
+    group: root                                                                 
+    mode: "0644"                                                                
+  when: dh_moduli.stat.checksum == '2dc7b3b18a2d3884ca4604466b75cf7903e96a97' 
```

Comment:
There's no point really in using a cryptographic hash function here at all as this file has come from a signed GPG package--we just need a (non-cryptographic) hash function to see if it's already been modified or not for idempotency. I think this was the wrong way to do it as if there are updates to this file provided by a new version of openssh it will no longer function correctly. The first function could be a call to awk with the inequality reversed (i.e., `command: awk '$5 > 2000' /etc/ssh/moduli`), and the following two functions would be predicated on `when: has_small_primes.stdout`.